client/network/src/service.rs

```diff
@@ -879,11 +879,14 @@ impl Metrics {
 				},
 				&["protocol"]
 			)?, registry)?,
-			notifications_total: register(CounterVec::new(
-				Opts::new(
-					"sub_libp2p_notifications_total",
-					"Number of notification received from all nodes"
-				),
+			notifications_sizes: register(HistogramVec::new(
+				HistogramOpts {
+					common_opts: Opts::new(
+						"sub_libp2p_notifications_sizes",
+						"Sizes of the notifications send to and received from all nodes"
+					),
+					buckets: vec![64.0, 256.0, 1024.0, 4.0 * 1024.0, 64.0 * 1024.0, 256.0 * 1024.0, 1024.0 * 1024.0],
```

Comment:
```suggestion
					buckets: vec![64.0, 256.0, 1024.0, 4.0 * 1024.0, 16.0 * 1024.0, 64.0 * 1024.0, 256.0 * 1024.0, 1024.0 * 1024.0],
```

Aren't you missing this one here?

Also why not use https://docs.rs/prometheus/0.8.0/prometheus/fn.exponential_buckets.html?